File: 0014-longest-common-prefix/longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        i = 0
        matchingChars = True
        while matchingChars:
            curr_char = ""
            for string in strs:
                if i >= len(string) or (curr_char != "" and string[i] != curr_char):
                    matchingChars = False
                    return string[:i]
                
                else:
                    curr_char = string[i]
            
            i += 1
                    
        return string

File: 0014-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string common_substr = "";

        for (int i = 0; i < strs[0].size(); ++i) {
            bool commonChar = true;
            for (int j = 0; j < strs.size() - 1; ++j) {
                if (i >= strs[j].size() || i >= strs[j + 1].size() || (strs[j].at(i) != strs[j + 1].at(i)))
                    return common_substr;
            }
            if (commonChar)
                common_substr.push_back(strs[0].at(i));   
        }
        return common_substr;
    }
};